BUG: TypeError when doing equality comparison with Extension Dtype

ataturhan21 opened the pull request:
- #43038 closes #xxxx
- 1 tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

ataturhan21 committed:
test added for issue #43038

phofl committed:
PERF: Improve performance in read_csv with numeric index col (#44610)

phofl committed:
BUG: Series.reset_index not ignoring name with inplace and drop (#44578)

Mikhaylov-yv committed:
[DOC] Transferring data from the 6th column to the first.  #44379 (#44427)

jorisvandenbossche committed:
PERF: improve find_common_type perf for special case of all-equal dtypes (#44594)

arushi-08 committed:
DOC: update shared_docs to clarify groupby with lists (#44569)

jbrockmendel committed:
BUG: concat_compat with 2D PeriodArray (#44598)

jbrockmendel committed:
REF: remove always-True DataFrame._iset_item kwarg (#44587)

jbrockmendel committed:
BUG: Series.where with IntervalDtype when no-op (#44585)

mzeitlin11 committed:
REF: use base class for BooleanArray any/all (#44549)

jonashaag committed:
Fix 'rtol' and 'atol' for numeric extension types (#44488)

feefladder committed:
Pass through Engine kwargs in ExcelWriter (#43445)

jbrockmendel committed:
BUG: DataFrame.shift with axis=1 and mismatched fill_value (#44564)

mroeschke committed:
DOC: GL01 Numpydoc validation (#44567)

johannes-mueller committed:
BUG: join MultiIndex with overlapping IntervalIndex level (#44096) (#44588)

loicdiridollou committed:
BUG: parse_dates=False while passing date_parser tries to use date parser (#44599)

* BUG: Providing fix for GH#44366

* BUG: fix pre-commit run

* BUG: adding other API changes for change of default argument

* Rewriting if statement and adding test

* BUG: changing pd.Series to Series to validate pre-commit hook

* Removing the second whatsnew statement only to keep the bug fix one

* BUG: simplifying the test with parser and parse_dates False #44366

* BUG: fixing pre-commit #44366

jbrockmendel committed:
BUG: Series.view from int64 to datetimelike (#44581)

pep8speaks commented:
Hello @fotino21! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/tests/dtypes/test_common.py`](https://github.com/pandas-dev/pandas/blob/5df8f47591d1f9a3a85e7ddcda4c0ee9ea4be82f/pandas/tests/dtypes/test_common.py):

> [Line 395:1](https://github.com/pandas-dev/pandas/blob/5df8f47591d1f9a3a85e7ddcda4c0ee9ea4be82f/pandas/tests/dtypes/test_common.py#L395): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

ataturhan21 committed:
Update test_common.py

jorisvandenbossche committed:
ASV: reduce overall run time for GroupByMethods benchmarks (#44604)

loicdiridollou committed:
DOC: 'MS' offset alias does not work as described with pd.date_range() (#44534)

jreback commented on pandas/tests/dtypes/test_common.py:
```diff
@@ -391,6 +392,41 @@ def test_is_unsigned_integer_dtype(dtype):
     assert not com.is_unsigned_integer_dtype(dtype)
 
+def test_all_type_comparison():
+    assert pd.api.types.pandas_dtype("int_") == "int_"
```
pls paramterize instead of writing out like this

fotino21 replied:
I made the change you requested

jorisvandenbossche committed:
TST: make tests involving use_numexpr more robust (#44601)

TheDerivator committed:
ENH: option to change the number in `_dir_additions_for_owner` (#44335)

MarcoGorelli committed:
TYP: Signature of "reindex" incompatible with supertype "NDFrame" (#40984)

phofl committed:
BUG: read_fwf not handling skiprows correctly with iterator (#44621)

MarcoGorelli committed:
TYP: Signature of "rename" incompatible with supertype "NDFrame" (#40979)

johannes-mueller committed:
BUG: fix get_indexer_non_unique() with 'object' targets with NaNs (#44482) (#44483)

phofl committed:
BUG: read_csv not replacing nans before date conversion (#44620)

* BUG: read_csv not replacing nans before date convrsion

* Remove mypy ignore

* Remove init file

* Add comment

* Add returns section

jbrockmendel committed:
BUG: Series.mode with dt64tz or PeriodDtype (#44582)

mroeschke committed:
DOC: PR08 doctring validation (#44622)

mroeschke committed:
ENH: Add numba engine to rolling/expanding.std/var (#44461)

jbrockmendel committed:
CLN: remove 2012-era commented-out excel test (#44636)

jbrockmendel committed:
Address libaggregations FIXME (#44635)

jbrockmendel committed:
CLN/TST: address FIXMEs (#44637)

phofl committed:
DEP: Deprecate non boolean sort for concat (#44629)

phofl committed:
DOC: Improve userguide for index_col and usecols in read_csv (#44643)

phofl committed:
BUG: read_csv not applying dtype to index col (#44632)

ataturhan21 committed:
Update test_common.py

ataturhan21 reviewed (commented):
I made the change you requested @jreback

phofl committed:
DOC: Clarify how to keep lines with too many fields (#44652)

jbrockmendel committed:
CLN: dont depend on fixture implementation details (#44645)

mroeschke committed:
CI/TST: Change id to is for flaky tests (#44650)

jbrockmendel committed:
REF: EA quantile logic to EA._quantile (#44412)

phofl committed:
BUG: read_csv not recognizing bad lines with names given (#44646)

phofl committed:
BUG: read_csv not converting original col with keep_date_col true (#44633)

jbrockmendel committed:
TST: skip->xfail (#44647)

phofl committed:
BUG: read_csv raising error for python engine when names longer than header but equal to data (#44654)

phofl committed:
Typ numerical ops (#44607)

phofl committed:
BUG: read_csv raising ParserError when some chunks have less columns than header (#44644)

prithvijitguha committed:
TYP: changed variable new_pd_index to final_pd_index  (#44500)

jbrockmendel committed:
TST: stop skipping 1986 tests in test_eval (#44649)

phofl committed:
BUG: mangle_dup_cols in read_csv replacing existing cols when conflict with target col (#44641)

phofl committed:
TYP: Typ part of python_parser (#44406)

rth committed:
ENH Remove import time warning for missing lzma (#43495)

phofl committed:
WARN: Clarify future warnings for read_csv (#44651)

* WARN: Clarify future warnings for read_csv

* Adjust test

* Use namedtuple

* Fix tests

Bklyn committed:
Holiday rename; Juneteenth `start_date` (#44576)

* add Juneteenth to USFederalHolidayCalendar

* Standardize federal holiday names

Reference https://www.opm.gov/policy-data-oversight/pay-leave/federal-holidays/#url=2021

* short long code line

* ENH: Holidays: add Juneteenth start_date (#42328, #44574)

Co-authored-by: Chuliang Xiao <ChuliangX@gmail.com>
Co-authored-by: Jeff Reback <jeff@reback.net>

jbrockmendel committed:
ENH: __array_ufunc__ handle np.minimum.reduce (#43923)

gabrieldi95 committed:
testing broadcast on multiindex (#44094)

simonjayhawkins committed:
TYP: use TYPE_CHECKING for import_optional_dependency("numba") (#44273)

jbrockmendel committed:
TST: de-duplicate test_eval (#44661)

jbrockmendel committed:
TST: de-xfail (#44659)

lithomas1 committed:
COMPAT: Fix FutureWarning in tests with pyarrow 6.0.0 (#44662)

jbrockmendel committed:
Revert "TST: de-xfail (#44659)" (#44665)

This reverts commit 7b78b730aee122bb7d6e90723c4208da234de19e.

ivanovmg committed:
DOC: move info docs to DataFrameInfo (#38062)

jorisvandenbossche committed:
ASV: remove duplicated SelectDtypes benchmark (#44668)

jorisvandenbossche committed:
ASV: time_reindex_both_axes to reindex existing columns (#44667)

jbrockmendel committed:
BUG: groupby.sum with timedelta64 and NaT (#44658)

jreback commented:
@fotino21 thanks, tests appear to be failing

tobiasmcnulty committed:
DOC: Include a small preview of the groupby() user guide in its API reference (#43880)

ataturhan21 committed:
Update test_common.py

ataturhan21 committed:
Update test_common.py

mroeschke committed:
CI/TST: Add network tests to CI (#44648)

jbrockmendel committed:
TST/REF: fixturize TestEval (#44663)

fangchenli committed:
DEPS: bump sqlalchemy min support version to 1.4.0 (#43191)

ataturhan21 committed:
Update test_common.py

jorisvandenbossche committed:
PERF/BUG: ensure we store contiguous arrays in DataFrame(ndarray) for ArrayManager (#44562)

ataturhan21 committed:
Update test_common.py

phofl commented on pandas/tests/dtypes/test_common.py:
```diff
@@ -392,6 +392,48 @@ def test_is_not_unsigned_integer_dtype(dtype):
+        ("Float64", "Float64"),
+    ],
+)
+def test_all_type_comparison_with_true_result(dtype1, dtype2):
```
Your dtypes are always equal, so no need to use 2. Can simply use dtype1 for everything (also rename to dtype in this case)

fotino21 replied:
But there are some differences in case, is it not important?

phofl replied:
the test below is fine, but here everything is equal?

phofl commented on pandas/tests/dtypes/test_common.py:
```diff
@@ -392,6 +392,48 @@ def test_is_not_unsigned_integer_dtype(dtype):
+        ("Float64", "float64"),
+    ],
+)
+def test_all_type_comparison_with_false_result(dtype1, dtype2):
```
please call dtype and expected

fotino21 replied:
after changing dtype1 to dtype and dtype2 to expected, do i need to change anything else?

fotino21 replied:
@phofl I made the change you wanted, is there anything else to merge this pr?

jbrockmendel committed:
TST: fixturize test_eval (#44675)

ataturhan21 committed:
Update test_common.py

phofl commented on pandas/tests/dtypes/test_common.py:
```diff
@@ -392,6 +392,48 @@ def test_is_not_unsigned_integer_dtype(dtype):
+        ("Float64", "Float64"),
+    ],
+)
+def test_all_type_comparison_with_true_result(dtype, expected):
```
Please use only one variable here

fotino21 replied:
I didnt understand, can you explain what you mean?

phofl replied:
Dtype and expected is always the same. You can simply use dtype instead of both

fotino21 replied:
@pytest.mark.parametrize(
    "dtype",
    [
        ("int_"),
        ("int8"),
        ("Int8"),
        ("int16"),
        ("Int16"),
        ("int32"),
        ("Int32"),
        ("int64"),
        ("Int64"),
        ("uint8"),
        ("uint16"),
        ("uint32"),
        ("uint64"),
        ("float16"),
        ("float32"),
        ("Float32"),
        ("float64"),
        ("Float64"),
    ],
)
def test_all_type_comparison_with_true_result(dtype):
    assert pandas_dtype(dtype) == dtype


@pytest.mark.parametrize(
    "dtype,expected",
    [
        ("Int8", "int8"),
        ("Int16", "int16"),
        ("Int32", "int32"),
        ("Int64", "int64"),
        ("Float32", "float32"),
        ("Float64", "float64"),
    ],
)

fotino21 replied:
@phofl is it the change you want?

phofl replied:
Yes, but you can remove the brackets

fotino21 replied:
@pytest.mark.parametrize(
"dtype",
("int_"),
("int8"),
("Int8"),
("int16"),
("Int16"),
("int32"),
("Int32"),
("int64"),
("Int64"),
("uint8"),
("uint16"),
("uint32"),
("uint64"),
("float16"),
("float32"),
("Float32"),
("float64"),
("Float64"),
)
def test_all_type_comparison_with_true_result(dtype):
assert pandas_dtype(dtype) == dtype

@pytest.mark.parametrize(
"dtype,expected",
("Int8", "int8"),
("Int16", "int16"),
("Int32", "int32"),
("Int64", "int64"),
("Float32", "float32"),
("Float64", "float64"),
)

fotino21 replied:
like this?

phofl replied:
No, the round ones

phofl replied:
In the First test

fotino21 replied:
@pytest.mark.parametrize(
"dtype",
[
"int_",
"int8",
"Int8",
"int16",
"Int16",
"int32",
"Int32",
"int64",
"Int64",
"uint8",
"uint16",
"uint32",
"uint64",
"float16",
"float32",
"Float32",
"float64",
"Float64",
],
)
def test_all_type_comparison_with_true_result(dtype):
assert pandas_dtype(dtype) == dtype

@pytest.mark.parametrize(
"dtype,expected",
[
("Int8", "int8"),
("Int16", "int16"),
("Int32", "int32"),
("Int64", "int64"),
("Float32", "float32"),
("Float64", "float64"),
],
)

fotino21 replied:
is there any other problem?

ataturhan21 committed:
Update test_common.py

jbrockmendel committed:
CLN: TODOs and FIXMEs (#44683)

phofl commented on pandas/tests/dtypes/test_common.py:
```diff
@@ -392,6 +392,48 @@ def test_is_not_unsigned_integer_dtype(dtype):
+    ],
+)
+def test_all_type_comparison_with_true_result(dtype):
+    assert pandas_dtype(dtype) == dtype
```
Please add issue number

fotino21 replied:
added

ataturhan21 committed:
Update test_common.py

jbrockmendel committed:
BUG: Series.__rfloordiv__ with list/tuple (#44674)

jbrockmendel committed:
PERF: assert_frame_equal (#44695)

tswast committed:
TST: check compatibility with pyarrow types_mapper parameter (#44369)

ivanovmg committed:
ENH: enable Series.info() (#37320)

DriesSchaumont committed:
BUG: using .replace() on a Series containing np.nan raising ValueError when using regex=True. (#44653)

ataturhan21 committed:
tests added for dtype

fotino21 commented:
@phofl I made all the changes you requested and my latest commit passed all the tests. Is this pr ready to merge?

github-actions[bot] committed:
⬆️ UPGRADE: Autoupdate pre-commit config (#44701)

Co-authored-by: MarcoGorelli <MarcoGorelli@users.noreply.github.com>

mroeschke committed:
TST: Use IPython instance fixture to avoid file leak (#44700)

phofl committed:
Type read_fwf (#44678)

lithomas1 committed:
CI: Try to not auto-cancel on master (#44682)

jbrockmendel committed:
TST: fix Nullable xfails (#44685)

jbrockmendel committed:
REF: simplify Block.where (and subtle alignment bug) (#44691)

jbrockmendel committed:
BUG: IntegerArray/FloatingArray constructors mismatched NAs (#44514)

jbrockmendel committed:
TST: de-duplicate fixtures (#44699)

jbrockmendel committed:
TST: fix xfails/skips (#44706)

Mikhaylov-yv committed:
Removed link to Engarde project. It doesn't develop anymore. (#44708)

phofl committed:
Deprecate positional arguments for read_fwf (#44710)

mroeschke committed:
CI/TST: Use CI moto service instead of subprocess (#44612)

rlvoyer committed:
TST: Add test for where inplace (#44255)

twoertwein committed:
CLN: contextmanagers should yield in a try-finally block (#44716)

mroeschke committed:
TST: Fix local run on s3 tests (#44725)

jbrockmendel committed:
PERF: DataFrame.T (#44724)

fotino21 commented:
@jreback @phofl Is there anything else i need to do before merge?

jreback commented on pandas/tests/dtypes/test_common.py:
```diff
@@ -392,6 +392,51 @@ def test_is_not_unsigned_integer_dtype(dtype):
     assert not com.is_unsigned_integer_dtype(dtype)
 
 
+@pytest.mark.parametrize(
```
can you use the fixture instead look in pandas/conftests.py

jreback commented on pandas/tests/dtypes/test_common.py:
```diff
@@ -392,6 +392,51 @@ def test_is_not_unsigned_integer_dtype(dtype):
+@pytest.mark.parametrize(
+    "dtype,expected",
+    [
+        ("Int8", "int8"),
```
you can use the fixture here as well

fotino21 replied:
@pytest.fixture
def dtype():
 return [
        "int_",
        "int8",
        "Int8",
        "int16",
        "Int16",
        "int32",
        "Int32",
        "int64",
        "Int64",
        "uint8",
        "uint16",
        "uint32",
        "uint64",
        "float_",
        "float16",
        "float32",
        "Float32",
        "float64",
        "Float64",
    ]
def test_all_type_comparison_with_true_result(dtype):
    # GH#43038
    for element in dtype:
        assert pandas_dtype(element) == element



@pytest.fixture
def dtype_and_expected():
 return [ 
        ("Int8", "int8"),
        ("Int16", "int16"),
        ("Int32", "int32"),
        ("Int64", "int64"),
        ("Float32", "float32"),
        ("Float64", "float64"),
    ]
def test_all_type_comparison_with_false_result(dtype_and_expected):
    # GH#43038
    for element in dtype_and_expected:
        assert not pandas_dtype(element[0]) == element[1]

fotino21 replied:
@jreback is this the change you want?

jreback replied:
use the existing fixtures do not create new ones

fotino21 replied:
is there an example that uses these fixtures you say?

fotino21 replied:
def test_all_type_comparison_with_true_result(any_real_numpy_dtype):
    # GH#43038
    assert pandas_dtype(any_real_numpy_dtype) == any_real_numpy_dtype

fotino21 replied:
@jreback is this the thing i need to do?

jreback replied:
yes

jbrockmendel committed:
CLN: TODOs/FIXMEs (#44717)

jbrockmendel committed:
BUG: Series.where with incompatible NA value (#44697)

mroeschke committed:
DOC: validate_docstrings cleans up leftover files; doctest +SKIP file examples (#44711)

nickleus27 committed:
TYP: changed variable cat_array to cat_array_list in dtypes.py and arr_list in categorical.py (#44664)

jorisvandenbossche committed:
PERF: passthrough mask to take_1d if already known (#44666)

jorisvandenbossche committed:
PERF/TYP: typing cast in __getitem__ gives quite some overhead (#44624)

* PERF/TYP: typing cast in __getitem__ gives quite some overhead

* stringify

* categorical as well

jbrockmendel committed:
PERF: slow tests (#44727)

valentinoli committed:
DOC: fix examples in Timestamp functions (#44702)

* DOC: fix examples in Timestamp functions

`Timestamp.utcfromtimestamp()` and `Timestamp.fromtimestamp()`

* DOC: fix examples in Timestamp functions

`Timestamp.utcfromtimestamp()` and `Timestamp.fromtimestamp()`

phofl committed:
Deprecate prefix argument in read_csv and read_table (#44713)

jbrockmendel committed:
CLN: TODOs (#44733)

jbrockmendel committed:
TST: dont skip dt64tz rolling tests (#44730)

jbrockmendel committed:
TST: remove non-strict xfails (#44728)

phofl committed:
TST: read_csv raisin error with string dtype and parse dates (#44694)

jbrockmendel committed:
CI/DOC: warning in docbuild (#44756)

ChristofKaufmann committed:
DOC: Improve code example for DataFrame.join (#44735)

mroeschke committed:
TST: Parameterize test_algos (#44760)

mroeschke committed:
Fix compression misspelling in ExcelWriter (#44759)

rhshadrach committed:
REGR: find_stack_level reverts (#44753)

attack68 committed:
CLN: Todo test reductions in test_style.py (#44750)

johnzangwill committed:
ENH: Add indexing syntax to GroupBy.nth() (#44688)

jbrockmendel committed:
BUG: FloatingArray(float16data) (#44715)

jbrockmendel committed:
REF: share _reduce method (#44737)

jbrockmendel committed:
ENH: implement tm.shares_memory (#44747)

jbrockmendel committed:
BUG: series.interpolate(inplace=True) *actually* inplace (#44749)

jbrockmendel committed:
ENH: improved dtype inference for Index.map (#44609)

Varun270 committed:
Deprecate mapper argument in rename (#44672)

ataturhan21 committed:
Update test_common.py

ataturhan21 committed:
Update test_common.py

ataturhan21 committed:
fixture added

ataturhan21 committed:
Merge branch 'tests-for-dtype-comparisons' of https://github.com/fotino21/pandas into tests-for-dtype-comparisons

ataturhan21 committed:
fixtures added

ataturhan21 committed:
Merge branch 'tests-for-dtype-comparisons' of https://github.com/fotino21/pandas into tests-for-dtype-comparisons

ataturhan21 committed:
fix

phofl commented on pandas/tests/dtypes/test_common.py:
```diff
@@ -392,49 +392,15 @@ def test_is_not_unsigned_integer_dtype(dtype):
-    ],
-)
-def test_all_type_comparison_with_true_result(dtype):
+def test_all_type_comparison_with_true_result(any_real_numpy_dtype):
```
Could you also add the numeric ea dtypes?

fotino21 replied:
@phofl yes, i will add but my code doesnt pass all the tests because of some other files. What can i do for that?

=========================== short test summary info ============================
FAILED pandas/tests/arrays/floating/test_arithmetic.py::test_unary_float_operators[Float32-source0-neg_target0-abs_target0]
FAILED pandas/tests/arrays/floating/test_arithmetic.py::test_unary_float_operators[Float32-source1-neg_target1-abs_target1]
FAILED pandas/tests/arrays/floating/test_arithmetic.py::test_unary_float_operators[Float32-source2-neg_target2-abs_target2]
FAILED pandas/tests/arrays/floating/test_arithmetic.py::test_unary_float_operators[Float64-source0-neg_target0-abs_target0]
FAILED pandas/tests/arrays/floating/test_arithmetic.py::test_unary_float_operators[Float64-source1-neg_target1-abs_target1]
FAILED pandas/tests/arrays/floating/test_arithmetic.py::test_unary_float_operators[Float64-source2-neg_target2-abs_target2]
FAILED pandas/tests/arrays/floating/test_astype.py::test_astype_copy - Future...
FAILED pandas/tests/arrays/floating/test_comparison.py::TestComparisonOps::test_no_shared_mask[Float32Dtype]
FAILED pandas/tests/arrays/floating/test_comparison.py::TestComparisonOps::test_no_shared_mask[Float64Dtype]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int8-source0-neg_target0-abs_target0]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int8-source1-neg_target1-abs_target1]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int8-source2-neg_target2-abs_target2]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int16-source0-neg_target0-abs_target0]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int16-source1-neg_target1-abs_target1]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int16-source2-neg_target2-abs_target2]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int32-source0-neg_target0-abs_target0]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int32-source1-neg_target1-abs_target1]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int32-source2-neg_target2-abs_target2]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int64-source0-neg_target0-abs_target0]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int64-source1-neg_target1-abs_target1]
FAILED pandas/tests/arrays/integer/test_arithmetic.py::test_unary_int_operators[Int64-source2-neg_target2-abs_target2]
FAILED pandas/tests/arrays/integer/test_comparison.py::TestComparisonOps::test_no_shared_mask[Int8Dtype]
FAILED pandas/tests/arrays/integer/test_comparison.py::TestComparisonOps::test_no_shared_mask[Int16Dtype]
FAILED pandas/tests/arrays/integer/test_comparison.py::TestComparisonOps::test_no_shared_mask[Int32Dtype]
FAILED pandas/tests/arrays/integer/test_comparison.py::TestComparisonOps::test_no_shared_mask[Int64Dtype]
FAILED pandas/tests/arrays/integer/test_comparison.py::TestComparisonOps::test_no_shared_mask[UInt8Dtype]
FAILED pandas/tests/arrays/integer/test_comparison.py::TestComparisonOps::test_no_shared_mask[UInt16Dtype]
FAILED pandas/tests/arrays/integer/test_comparison.py::TestComparisonOps::test_no_shared_mask[UInt32Dtype]
FAILED pandas/tests/arrays/integer/test_comparison.py::TestComparisonOps::test_no_shared_mask[UInt64Dtype]
FAILED pandas/tests/arrays/integer/test_dtypes.py::test_astype_copy - FutureW...
= 30 failed, 122283 passed, 18661 skipped, 870 xfailed, 1 xpassed in 1386.44s (0:23:06) =

fotino21 replied:
@jreback @phofl My latest commit has passed all the tests, do i need to do anything else before merge?

ataturhan21 committed:
Update test_common.py

xiki-tempula committed:
ENH: allow attrs to be propagated via pd.concat (#42252)

jbrockmendel committed:
CLN/TST: skips -> xfails (#44744)

ataturhan21 committed:
Merge branch 'tests-ons' of https://github.com/fotino21/pandas into tests-for-dtype-comparisons

added fix



:Qa
:qw

ataturhan21 committed:
fixtured

skwirskj committed:
44414 data frame attributes not propagating with astype (#44680)

Varun270 committed:
Updated code_checks.sh with PR06 (#43511)

Carreau committed:
DOC: Uniform block directive syntax. (#44757)

MarcoGorelli committed:
fix PR06 failing CI (#44773)

jbrockmendel committed:
ENH: IntervalArray.min/max (#44746)

jbrockmendel committed:
CLN: FIXMEs (#44771)

ataturhan21 committed:
Merge branch 'tests-for-dtype-comparisons' of https://github.com/fotino21/pandas into tests-for-dtype-comparisons

ataturhan21 committed:
fixtures

jreback commented:
something when wrong

fotino21 commented:
> 

@jreback I didn't understand

jreback commented:
@fotino21 you are updating a huge amount of code. make sure you are merging in master

fotino21 commented:
> @fotino21 you are updating a huge amount of code. make sure you are merging in master

@jreback My changes are small and they are only in test_common file. What do i need to do for other commits by other people. And sorry for the inconvenience. This is my first contribution to a huge code base.

fotino21 commented:
@jreback @phofl Do I need to do anything else before merge?

jreback commented:
> @jreback @phofl Do I need to do anything else before merge?

yes you need to rebase against upstream master

fotino21 commented:
> > @jreback @phofl Do I need to do anything else before merge?
> 
> yes you need to rebase against upstream master

@jreback I made it following this link: https://timonweb.com/misc/how-to-update-a-forked-repo-from-an-upstream-with-git-rebase-or-merge/

fotino21 commented:
@jreback @phofl Do I need to do anything else before merge?

jreback commented:
@fotino21 this has included lots of other changes u need to merge upstream/master

fotino21 commented:
> @fotino21 this has included lots of other changes u need to merge upstream/master



> @fotino21 this has included lots of other changes u need to merge upstream/master

How can i do that? I already based my master branch to upstream/master

MarcoGorelli commented:
can you try
```
git fetch upstream
git rebase upstream/master
git push origin HEAD -f
```
?

fotino21 commented:
> @MarcoGorelli 

In the second step, I only see this message:

noop
# Rebase bc17343f93..bc17343f93 onto bc17343f93 (1 command)
#
# Commands:
# p, pick <commit> = use commit
# r, reword <commit> = use commit, but edit the commit message
# e, edit <commit> = use commit, but stop for amending
# s, squash <commit> = use commit, but meld into previous commit
# f, fixup [-C | -c] <commit> = like "squash" but keep only the previous
#                    commit's log message, unless -C is used, in which case
#                    keep only this commit's message; -c is same as -C but
#                    opens the editor
# x, exec <command> = run command (the rest of the line) using shell
# b, break = stop here (continue rebase later with 'git rebase --continue')
# d, drop <commit> = remove commit
# l, label <label> = label current HEAD with a name
# t, reset <label> = reset HEAD to a label
# m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
# .       create a merge commit using the original merge commit's
# .       message (or the oneline, if no original merge commit was
# .       specified); use -c <commit> to reword the commit message
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#

MarcoGorelli commented:
🤔  to be honest I don't know, I can't even tell how you opened this PR as it doesn't appear to have come from a branch

the other PR (https://github.com/pandas-dev/pandas/pull/44840) does come from a branch, so maybe it's OK to continue there, then I can check it out locally if necessary

jreback commented:
closing in favor of #44840 

@fotino21 ideallly we only have a single, original PR as it makes more work for maintaining.
